src/core/qgstilecache.cpp

```diff
@@ -33,16 +33,20 @@ void QgsTileCache::insertTile( const QUrl &url, const QImage &image )
 
 bool QgsTileCache::tile( const QUrl &url, QImage &image )
 {
+  QNetworkRequest req( url );
+  QgsNetworkAccessManager::instance()->preprocessRequest( &req );
```

Comment:
Why not handle this automatically in QgsNetworkAccessManager::createRequest ? then we wouldn't have to update ALL the code which creates requests to handle this...

Reply:
It is already automatically done in QgsNetworkAccessManager::createRequest. In QgsTileCache, it is called to get the url after preprocessing. Otherwise, if the preprocessing changes the url, those tiles would not be found in the cache.

Reply:
Agh, missed that!

Can I ask what the intended use case is for these preprocessors? Is it purely to alter the URL? If so, I wonder if a "URL preprocessor" is more appropriate than a "request preprocessor"...

Reply:
We use it for two things:
- One is indeed altering the URL to inject authentication tokens, but in priciple one might also want to inject a header
- The other one is debugging, i.e. a python plugin which dumps the entire request + headers

Reply:
> One is indeed altering the URL to inject authentication tokens, but in priciple one might also want to inject a header

Fair enough

> The other one is debugging, i.e. a python plugin which dumps the entire request + headers

Are you aware of the inbuilt network logger? You may want to adapt your plugin to use the same approach

Reply:
@mhugent 

> Otherwise, if the preprocessing changes the url, those tiles would not be found in the cache.

Mind adding a comment to the code explaining this too?